ui/src/app/pages/basePage.tsx

```diff
@@ -176,6 +183,11 @@ export abstract class PageComponent<P extends PageProps, S extends PageState> ex
         return this.state.isError ? true : false;
     }
 
+    private is403Error(): boolean {
+        return this.state.error && this.state.error.error && this.state.error.error.message &&
+            this.state.error.error.message.includes("403");
```

Comment:
there isn't a better way to detect that the http response has status 403?

Reply:
I know, right?  But I couldn't find a better way.  I inspected the response we get from Axios (which is the REST client we use) and they don't seem to separately expose the HTTP response code.  At least not that I could find experimentally.

Reply:
If you get a response from the server (which should be the case here), it should be available using `error.response.status`.

Reply:
Did you test that in the browser?  I printed out the entire error object in the browser and there wasn't any status property.

Reply:
Here is an example of the entire axios error object:

```json
{
   "message": "Request failed with status code 403",
   "name": "Error",
   "stack": "Error: Request failed with status code 403\n    at createError (webpack-internal:///./node_modules/axios/lib/core/createError.js:16:15)\n    at settle (webpack-internal:///./node_modules/axios/lib/core/settle.js:17:12)\n    at XMLHttpRequest.handleLoad (webpack-internal:///./node_modules/axios/lib/adapters/xhr.js:62:7)",
   "config": {
      "url": "http://localhost:8080/t/test-registry-1/apis/registry/v2/search/artifacts?limit=10&order=asc&orderby=name",
      "method": "get",
      "headers": {
         "Accept": "application/json",
         "Authorization": "Bearer **redacted**"
      },
      "transformRequest": [
         null
      ],
      "transformResponse": [
         null
      ],
      "timeout": 0,
      "xsrfCookieName": "XSRF-TOKEN",
      "xsrfHeaderName": "X-XSRF-TOKEN",
      "maxContentLength": -1,
      "maxBodyLength": -1
   }
}
```

Reply:
Oh snap - you are right.  There is a `response` property that gets ignored by `JSON.stringify()`.  I didn't know that was even possible.

Reply:
OK fixed.  Thanks for pointing this out.